adala/agents/base.py

```diff
@@ -429,12 +431,16 @@ async def arefine_skill(
             predictions = None
         else:
             inputs = InternalDataFrame.from_records(data or [])
-            predictions = await self.skills.aapply(inputs, runtime=runtime)
-
+            if reapply:
+                predictions = await self.skills.aapply(inputs, runtime=runtime)
+            else:
+                predictions = inputs
```

Comment:
in the future, it should be possible to automatically detect if predictions are missing and aapply in that case, and get rid of this `reapply` arg, right?

Reply:
actually it doesn't seem clear to me whether predictions must be applied or not. However, the performance/cost implications are clear. So I'd leave this option for now and put off to decide on the best setup later when we will have some benchmarks to understand the quality tradeoff.